File: gild/prime/prime.rb
#!/usr/bin/ruby

class Prime
  attr_accessor :limit ,:sum
  def initialize(limit='')
    @limit=limit.to_i
    @sum=0
    @allPrimes=[]
    (2..@limit).each do |num|
      if prime?(num)
        @sum+=num
        @allPrimes.push num
        puts num
      end
    end
    puts @sum
  end

  def prime?(number)
    limit=Math.sqrt(number)
    if number===1 or number === 0 or limit=== Integer(limit)
      return false
    end
    @allPrimes.each do |check|
      return true if check >limit
      if number%check===0
        return false
      end
    end
    true
  end
end

if __FILE__ == $0
  text=""
  file=File.new(ARGV[0],"r")
  key=file.gets
  prime=Prime.new(key.sub("\n",""))
end

File: gild/prime/prime.c
#include<stdio.h>
#include<math.h>
#include<stdlib.h>

int isPrime(long n){
  int limit=0,i;
  limit=sqrt(n);
  if(n==0||n==1)
    return 0;
  for(i=2;i<=limit;i++){
    if(n%i==0)
      return 0;
  }
  return 1;
}

int main(int argc, char *argv[]){
  FILE *file = fopen( argv[1], "r" );
  long long sum=0;
  long num,i;
  char len[20];

  fgets(len,10,file );
  num=atoi(len);

  for(i=2;i<=num;i++){
    if(isPrime(i)){
      sum+=i;
    }
  }
  printf("%llu\n",sum);
  fclose( file );
  return 0;
}
